File: personal_project_front-end/src/models/request/RoomConditionRequest.ts
import { RoomStatusEnum } from "@/enums/RoomStatusEnum";
import {
  defaultPagingQueryConditionRequest,
  PagingQueryConditionRequest,
} from "./PagingQueryConditionRequest";

export type RoomConditionRequest = PagingQueryConditionRequest & {
  status?: RoomStatusEnum;
};

export const defaultRoomConditionRequest: RoomConditionRequest = {
  ...defaultPagingQueryConditionRequest,
};

File: personal_project_back-end/main-service/src/main/java/com/example/mainservice/models/requests/RoomConditionRequest.java
package com.example.mainservice.models.requests;

import com.example.common.service.enums.RoomStatusEnum;
import com.example.common.service.models.requests.PagingQueryConditionRequest;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class RoomConditionRequest extends PagingQueryConditionRequest {
    RoomStatusEnum status;
}
